pkg/runtime/reconciler.go

```diff
@@ -257,6 +257,8 @@ func (r *resourceReconciler) Reconcile(ctx context.Context, req ctrlrt.Request)
 		if err != nil {
 			return r.handleCacheError(ctx, err, desired)
 		}
+	} else if !needCARMLookup && acctID == "" {
+		return ctrlrt.Result{}, fmt.Errorf("no default account id nor account id in namespace found")
 	}
```

Comment:
Inexistant accountID is a symptom is a consequence of credentials not being provided. tiny suggestion:
```suggestion
	} else if !needCARMLookup && acctID == "" {
		return ctrlrt.Result{}, fmt.Errorf("unable to locate AWS credentials to manage resource: both controller credentials and a proper CARM configuration are missing")
	}
```